heroic-tattoo.lic

```diff
@@ -0,0 +1,57 @@
+custom_require.call(%w[common])
+
+no_pause_all
+no_kill_all
+
+no_use_rooms = get_settings.wand_watcher_no_use_rooms
+no_use_scripts = ['burgle','go2','athletics']
+downtime = DRSkill.getrank('Arcana') < 1400 ? (60 - DRSkill.getrank('Arcana')/24)*60 : 0
+timer = 3600 + downtime
+next_invoke = Time.now
+
+arg_definitions = [[
+      { name: 'once', regex: /once/i, optional: true, description: "One activation, however long that takes"},
+      { name: 'try', regex: /try/i, optional: true, description: "One ATTEMPT at activation, regardless of success or failure, and does not continue attempting"},
+      { name: 'cycle', regex: /cycle/i, optional: true, description: "Outputs information about your tattoo cooldown, uptime, downtime, cycle time" },
+      { name: 'script_summary', optional: true, description: "Script activates and maintains heroic tattoo buffs, running in the background. Best used as an autostart."}
+]]
+args = parse_args(arg_definitions)
+if args.cycle
+    DRC.message("Heroic Tattoo uptime is 60 minutes per activation")
+    DRC.message("Approximate downtime on your Heroic Tattoo is : #{downtime/60} minutes")
+    DRC.message("So total cycle from activation to next possible activation is #{(downtime/60)+60} minutes")
+    exit
+end
+
+pause 5
+loop do
+    while next_invoke > Time.now || no_use_scripts.any? { |name| Script.running?(name) } || ( no_use_rooms.any? { |room| room === DRRoom.title.to_s()[2..-3] || room == Room.current.id} )
+        pause 15
+    end
+    scripts = DRC.smart_pause_all
+
+    case DRC.bput("invoke my tattoo",
+        'as its magic washes over you',
+        'Its magic appears depleted',
+        'already under a heroic effect',
+        'Closing your eyes, you carefully bend')
+    when 'as its magic washes over you'
+        next_invoke = Time.now + timer
+        DRC.message("Next attempt in #{(next_invoke - Time.now).to_i/60} minutes")
+        break if args.once
```

Comment:
Are we properly unpausing the scripts that were paused when we `break` for args.once?

Reply:
Answering my own question: I think we do. It's a break not an exit. Thanks.

Reply:
actually, this is a problem, it leaves the whole loop. works on try, not on once.